### PR TITLE
Feat/cost reduction

### PR DESCRIPTION
**Pull request by ioanaroceanu2000**
Functions added
- submit batch
- get fee (per deposit)
- get batch budget
- get gas fee of submit batch

### CODE REVIEW

File: src/entities/mellow/mellowLpRouter.ts [hunk 1]
```diff
@@ -821,6 +832,124 @@ class MellowLpRouter {
...
+    } catch (err) {
+      const sentryTracker = getSentryTracker();
+      sentryTracker.captureException(err);
+      sentryTracker.captureMessage('Failed to get batch budget');
+      console.error('Error while getting batch budget', err);
+      throw new Error('Failed to get batch budget');
+    }
+  };
+
+  getBatchBudgetEth = async (): Promise<BigNumber> => {
```
Comment (ssbarbee) [inline src/entities/mellow/mellowLpRouter.ts:914]:
```suggestion
  getBatchBudgetUnderlyingToken = async (): Promise<BigNumber> => {
```

File: src/entities/mellow/mellowLpRouter.ts [hunk 2]
```diff
@@ -821,6 +832,124 @@ class MellowLpRouter {
 
     return this.canManageVaultPositions[vaultIndex];
   };
+
+  submitBatch = async (): Promise<ContractReceipt> => {
```
Comment (avniculae) [inline src/entities/mellow/mellowLpRouter.ts:836]: I would rename this `submitAllBatchesForFee`

File: src/entities/mellow/mellowLpRouter.ts [hunk 3]
```diff
@@ -821,6 +832,124 @@ class MellowLpRouter {
...
+  getSubmitBatchGasCost = async (): Promise<BigNumber> => {
+    if (isUndefined(this.writeContracts)) {
+      throw new Error('Uninitialized contracts.');
+    }
+
+    try {
+      const gasUnitsEstimate =
+        await this.writeContracts.mellowRouter.estimateGas.submitAllBatchesForFee();
+      // convert gas estimate from gas units into usd
+      const gasUnitPriceUSD = await convertGasUnitsToUSD(this.provider, 1);
```
Comment (avniculae) [inline src/entities/mellow/mellowLpRouter.ts:881]: you already have this as a member in our class

File: src/entities/mellow/mellowLpRouter.ts [hunk 4]
```diff
@@ -821,6 +832,124 @@ class MellowLpRouter {
...
+
+  getBatchBudgetEth = async (): Promise<BigNumber> => {
+    if (isUndefined(this.writeContracts)) {
+      throw new Error('Uninitialized contracts.');
+    }
+
+    try {
+      let remainingDeposits = 0;
+      for (let i = 0; i < this.vaultsCount; i++) {
+        remainingDeposits += (await this.writeContracts.mellowRouter.getBatchedDeposits(i)).length;
```
Comment (0xZenus) [inline src/entities/mellow/mellowLpRouter.ts:922]: use readOnlyContracts, not writeContracts

File: src/entities/mellow/mellowLpRouter.ts [hunk 5]
```diff
@@ -821,6 +832,124 @@ class MellowLpRouter {
...
+    }
+  };
+
+  getBatchBudgetUsd = async (): Promise<number> => {
+    if (isUndefined(this.writeContracts)) {
+      throw new Error('Uninitialized contracts.');
+    }
+
+    try {
+      const budgetEth = await this.getBatchBudgetEth();
```
Comment (avniculae) [inline src/entities/mellow/mellowLpRouter.ts:900]: budget is in underlying token. why is it called budgeteth?

File: src/entities/mellow/mellowLpRouter.ts (hunk 1 above)
Comment (avniculae) [inline src/entities/mellow/mellowLpRouter.ts:914]: why is it called budgetEth?

Comment (ioanaroceanu2000) [reply src/entities/mellow/mellowLpRouter.ts]: change to budget underlying [a09f7f2](https://github.com/Voltz-Protocol/v1-sdk/pull/164/commits/a09f7f220a9a2e4d81d125cf7c6c0b89fb49a6f5)

File: src/entities/mellow/mellowLpRouter.ts (hunk 5 above)
Comment (ioanaroceanu2000) [reply src/entities/mellow/mellowLpRouter.ts]: change to budget underlying [a09f7f2](https://github.com/Voltz-Protocol/v1-sdk/pull/164/commits/a09f7f220a9a2e4d81d125cf7c6c0b89fb49a6f5)

File: src/entities/mellow/mellowLpRouter.ts (hunk 3 above)
Comment (ioanaroceanu2000) [reply src/entities/mellow/mellowLpRouter.ts]: true, changed [a09f7f2](https://github.com/Voltz-Protocol/v1-sdk/pull/164/commits/a09f7f220a9a2e4d81d125cf7c6c0b89fb49a6f5)

File: src/entities/mellow/mellowLpRouter.ts (hunk 2 above)
Comment (ioanaroceanu2000) [reply src/entities/mellow/mellowLpRouter.ts]: renamed [a09f7f2](https://github.com/Voltz-Protocol/v1-sdk/pull/164/commits/a09f7f220a9a2e4d81d125cf7c6c0b89fb49a6f5)

File: src/entities/mellow/mellowLpRouter.ts (hunk 4 above)
Comment (ioanaroceanu2000) [reply src/entities/mellow/mellowLpRouter.ts]: changed [a09f7f2](https://github.com/Voltz-Protocol/v1-sdk/pull/164/commits/a09f7f220a9a2e4d81d125cf7c6c0b89fb49a6f5)

File: src/entities/mellow/mellowLpRouter.ts (hunk 1 above)
Comment (ioanaroceanu2000) [reply src/entities/mellow/mellowLpRouter.ts]: changed [a09f7f2](https://github.com/Voltz-Protocol/v1-sdk/pull/164/commits/a09f7f220a9a2e4d81d125cf7c6c0b89fb49a6f5)

**Review by avniculae — APPROVED** [review]

**Pull request merged**